modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuBarSkin.java

```diff
@@ -304,10 +304,12 @@ public MenuBarSkin(final MenuBar control) {
                 // RT-23147 when MenuBar's focusTraversable is true the first
                 // menu will visually indicate focus
                 unSelectMenus();
-                menuModeStart(0);
-                openMenuButton = ((MenuBarButton)container.getChildren().get(0));
-                setFocusedMenuIndex(0);
-                openMenuButton.setHover();
+                if (!container.getChildren().isEmpty()) {
```

Comment:
wondering whether this would be an appropriate time to simplify the logic a bit: as unSelectMenus is called in both if and else block, it could be condensed to

    unSelectMenus();
    if (t1 && !container.getChildren().isEmpty()) {
       ...
    }  

might overlook something, though

Reply:
Good suggestion. Fixed.